src/inference/basics.rs

```diff
@@ -283,6 +283,7 @@ impl FOModule {
                     return TransCexResult::Cancelled;
                 }
                 let resp = solver.check_sat(assumptions).expect("error in solver");
+                solver.save_tee();
```

Comment:
Can you move this to after the `match resp` so that it captures `solver.get_minimal_model()` as well?